docs/enhancements/infrastructure-monitoring-service-api/alarms.md

```diff
@@ -0,0 +1,770 @@
+              "openshift_io_alert_source": "platform",
+              "prometheus": "openshift-monitoring/k8s",
+              "severity": "info",
+              "upstream": "<default>"
+          },
+          "annotations": {
+              "description": "For more information refer to 'oc adm upgrade' or https://console-openshift-console.apps.cnfdf27.sno.telco5gran.eng.rdu2.redhat.com/settings/cluster/.",
+              "summary": "Your upstream update recommendation service recommends you update your cluster."
+          },
+          "startsAt": "2024-08-28T11:39:17.958Z",
+          "endsAt": "0001-01-01T00:00:00Z",
+          "generatorURL": "https://console-openshift-console.apps.cnfdf27.sno.telco5gran.eng.rdu2.redhat.com/monitoring/graph?g0.expr=sum+by+%28channel%2C+namespace%2C+upstream%29+%28cluster_version_available_updates%29+%3E+0&g0.tab=1",
+          "fingerprint": "91406cd113ad87e5"
+      },
+ ]
+```
+
+Equivalent AlarmEventRecord in Go 
+
+```go
+alarmEventRecord := AlarmEventRecord{
+		AlarmDefinitionID: "82139b9c-3683-427b-a1f5-4bf9d6dbb0d8", // See section "For a given ResourceTypeID and AlarmName (coming from AM alert), find the AlarmDefinitionID and ProbableCauseID"
+		ProbableCauseID:   "7fe31219-c2a2-48c7-a2f5-f4bcfe0486e7", // See section "For a given ResourceTypeID and AlarmName (coming from AM alert), find the AlarmDefinitionID and ProbableCauseID"
+        AlarmEventRecordID: uuid.New(), // Auto generated with DB                 
+		ResourceTypeID:     "c9d3f9c5-8429-4484-8179-2a7977071bbf", // See section "Notes on Init phase" on how labels.managed_cluster mapped to resourceTypeID
+		ResourceID:        "d43bf16b-c9c6-432b-934e-7b670cc6a2cc", // labels.managed_cluster
+		AlarmRaisedTime:   "12-31-1920 12:32:14Z",                 // comes from alertmanager `startsAt`
+        AlarmAcknowledged: false, 
+        PerceivedSeverity: SeverityCritical,                      // comes from labels.severity
+        Extensions:        []KeyValue{{Key: "namespace", Value: "openshift-cluster-version"}, ....}, // any labels that are not processed already (e.g skip labels.severity)
+
+        // Optionally 
+        AlarmChangedTime         // use changedAt
+        AlarmClearedTime         // use endsAt or potentially current time if we missed the resolved notification
+        AlarmAcknowledgedTime    // Done through API request AlarmEventRecordModifications.alarmAcknowledged (update AlarmAcknowledged)
+}
+```
+
+## Schema
+All O-RAN services will use the same O-CLOUD DB service. More on DB deployment [here](#postgres).
+
+Each table is modeled after O-RAN data structures. DB in our case wil be called `o-ran-infrastructure-monitoring-alarms`.
+Init SQL may look like the following:
+```sql
+CREATE DATABASE o-ran-infrastructure-monitoring-alarms;
+
+-- ENUM for ManagementInterfaceID
+DROP TYPE IF EXISTS ManagementInterfaceID CASCADE;
+CREATE TYPE ManagementInterfaceID AS ENUM ('O1', 'O2DMS', 'O2IMS', 'OpenFH');
```

Comment:
DB enums tend to be a bit difficult to deal with over schema upgrades if values need to be added.   Projects often use just regular integer columns and let the ORM map from enums in code to integers in DB.  This also has the side effect of consuming less space in the DB which is especially beneficial on large tables.

Reply:
Yeah good point about having issues with future migration. I'll go with the lookup table method instead of enum. Will update the doc. 

e.g 

```
CREATE TABLE management_interfaces (
    id SERIAL PRIMARY KEY,
    name VARCHAR(255) UNIQUE NOT NULL
);

INSERT INTO management_interfaces (name) VALUES ('O1'), ('O2DMS'), ('O2IMS'), ('OpenFH');
```

```
CREATE TABLE alarms_table (
    ...
    management_interface_id INT REFERENCES management_interfaces(id) NOT NULL DEFAULT 3
);
```

Reply:
Yes, that would work, but is it really necessary?  ORM's are usually able to do that mapping in code without requiring a separate table... but either way works.

Reply:
ok, but could probably still do this in code simply enough with the extra table.  Anyway... details.